### ISSUE TITLE
Nuitka does not detect MinGW64 installation automatically

### ISSUE THREAD

**Issue by Pro100rus32** [Nuitka/Nuitka#863]
1) 0.6.9.3
Python: 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
Executable: C:\Users\andre\AppData\Local\Programs\Python\Python38\python.exe
OS: Windows
Arch: x86_64
2. I downloaded nuitka.
I downloaded and installed mingw64 in the correct path, chose both i686 and x86_64, there is no difference.

Output:
```
C:\Users\andre>python -m nuitka --mingw64 --plugin-enable=multiprocessing "E:\NEWEXEC2\NAIMBeta comp.py"
Nuitka:WARNING:Not recursing to 'cv2' (C:\Users\andre\AppData\Local\Programs\Python\Python38\lib\site-packages\cv2), please specify --nofollow-imports (do not warn), --follow-imports (recurse to all), --nofollow-import-to=cv2 (ignore it), --follow-import-to=cv2 (recurse to it) to change.
Nuitka:WARNING:Not recursing to 'mouse' (C:\Users\andre\AppData\Local\Programs\Python\Python38\lib\site-packages\mouse), please specify --nofollow-imports (do not warn), --follow-imports (recurse to all), --nofollow-import-to=mouse (ignore it), --follow-import-to=mouse (recurse to it) to change.
Nuitka:WARNING:Not recursing to 'numpy' (C:\Users\andre\AppData\Local\Programs\Python\Python38\lib\site-packages\numpy), please specify --nofollow-imports (do not warn), --follow-imports (recurse to all), --nofollow-import-to=numpy (ignore it), --follow-import-to=numpy (recurse to it) to change.
Nuitka:WARNING:Not recursing to 'mss' (C:\Users\andre\AppData\Local\Programs\Python\Python38\lib\site-packages\mss), please specify --nofollow-imports (do not warn), --follow-imports (recurse to all), --nofollow-import-to=mss (ignore it), --follow-import-to=mss (recurse to it) to change.
Nuitka-Plugins:INFO: multiprocessing: Injecting pre-module load code for module 'multiprocessing':
Nuitka-Plugins:INFO: multiprocessing:     Monkey patching "multiprocessing" load environment.
Nuitka-Plugins:INFO: multiprocessing: Injecting post-module load code for module 'multiprocessing':
Nuitka-Plugins:INFO: multiprocessing:     Monkey patching "multiprocessing" for compiled methods.
Nuitka-Scons:INFO: Backend C compiler: gcc (gcc).
Error, cannot locate suitable C compiler. You have the following options:

a) If a suitable Visual Studio version is installed, it will not be located
   automatically, unless you install pywin32 for the Python installation
   below "C:\Users\andre\AppData\Local\Programs\Python\Python38".

b) To make it find Visual Studio without registry execute from Start Menu
   the 'Visual Studio Command Prompt' or "vcvarsall.bat". That will add
   Visual Studio to the "PATH". And it then will be detected.

c) Install MinGW64 to "C:\MinGW64" or "\MinGW", where then it is automatically
   detected or add it to PATH before executing Nuitka. But be sure to pick the
   proper variant (32/64 bits, your Python arch is 'x86_64'), or else cryptic errors
   will be shown.

Normal MinGW will not work! MinGW64 does not mean 64 bits, just better
Windows compatibility.
Cygwin based gcc will not work.
MSYS2 based gcc will not work.
clang-cl will only work if MSVC already worked.
```
Doesn't work with any code.

**Comment by kayhayen** [comment]
Just read what you quoted.

**Comment by Pro100rus32** [comment]
> Just read what you quoted.


This is my first time here, what's wrong?

**Comment by Pro100rus32** [comment]
```
C:\Users\andre>C:\\MinGW64\\mingw64\\bin\\gcc.exe --version
gcc.exe (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


C:\Users\andre>python -m nuitka --version
0.6.9.3
Python: 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]
Executable: C:\Users\andre\AppData\Local\Programs\Python\Python38\python.exe
OS: Windows
Arch: x86_64
```

**Comment by kayhayen** [comment]
Got you. I was the one not reading your issue description. That is supposed to work. I will check it out. In the mean time set CC environment variable to that mingw64 GCC exe path as a workaround.

**Comment by Pro100rus32** [comment]
> Got you. I was the one not reading your issue description. That is supposed to work. I will check it out. In the mean time set CC environment variable to that mingw64 GCC exe path as a workaround.


Are you talking about this?
![image](https://user-images.githubusercontent.com/64806740/96463300-4082aa00-122f-11eb-8508-9d08ddbf11a2.png)

**Comment by kayhayen** [comment]
Please try https://nuitka.net/doc/factory.html where I have fixed the issue.

Seems that when updating Scons, one of our patches got lost, the one which helps to find the MinGW64 automatically. I added that change, and this is likely hotfix material, with it being an regression. This initially only affected Python3 I assume, but we use that version for Python 2.7 now, so it's almost all supported versions.

This should restore the out of the box experience a bit.

**Comment by Pro100rus32** [comment]
Thanks! Everything is working)

**Comment by kayhayen** [comment]
Released as part of 0.6.9.5
